main.cpp

```diff
@@ -1,4 +1,9 @@
 #include "nethogs.cpp"
+#include <fcntl.h>
+#include <vector>
+
+std::pair<int,int> self_pipe = std::make_pair(-1, -1);
```

Comment:
The reason for this pipe is to be able to trigger the select loop from the interrupt handler, right? Might be worth adding a comment about that.